test/remote-connector.test.js

```diff
@@ -87,3 +87,81 @@ describe('RemoteConnector', function() {
     });
   });
 });
+
+describe('Custom Path', function() {
+
+  var test = this;
+
+  before(function(done) {
+
+    test.server = server = loopback();
+    server.set('legacyExplorer', false);
+    server.use('/api', loopback.rest());
+    server.set('host', '127.0.0.1');
+    server.set('port', 3001);
+    server.set('restApiRoot', '/api');
+    
+    var db_config = {
+      connector: loopback.Memory,
+      name: 'db'
+    };
+    var db = loopback.createDataSource('db', db_config);
+
+    var ServerModel_config = {
+      dataSource: 'db',
+      http: {
+        path: '/custom'
+      }
+    };
+
+    var ServerModel = loopback.PersistedModel.extend('TestModel', {foo: 'string'}, ServerModel_config);
+    ServerModel.attachTo(db);
+
+    server.dataSource('db', db_config);
+
+    server.model(ServerModel);
+
+    test.client = client = loopback();
+    client.set('legacyExplorer', false);
+    client.use('/api', loopback.rest());
+    client.set('host', '127.0.0.1');
+    client.set('port', 3000);
+    client.set('restApiRoot', '/api');
+
+    var remoteDS_config = {
+      connector: Remote,
+      url: 'http://127.0.0.1:3001/api',
+      name: 'remoteDS'
+    };
+    var remoteDS = loopback.createDataSource('remoteDS', remoteDS_config);
+
+    RemoteModel_config = {
+      dataSource: 'remoteDS',
+      http: {
+        path: '/custom'
+      }
+    };
+
+    var RemoteModel = loopback.PersistedModel.extend('TestModel', {foo: 'string'}, RemoteModel_config);
+    RemoteModel.attachTo(remoteDS);
+
+    client.dataSource('remoteDS', remoteDS_config);
+
+    client.model(RemoteModel);
+
+    done();
+
+  });
+
+  it('should support http.path configuration', function(done) {
+    var server = test.server;
+    var client = test.client;
+
+    server.listen(function() {
+      client.models.TestModel.create({foo: 'bar'}, function(err, instance) {
+        if (err) return assert(err);
```

Comment:
`if (err) return done(err)` is much better, as it will preserve the original stack trace.